libs/ballot-interpreter-vx/src/utils/images.ts

```diff
@@ -1,16 +1,103 @@
+import { safeParseInt } from '@votingworks/types';
+import { assert } from '@votingworks/utils';
 import { Buffer } from 'buffer';
-import { createCanvas, createImageData, loadImage } from 'canvas';
-import { createWriteStream } from 'fs';
+import { createCanvas, createImageData, ImageData, loadImage } from 'canvas';
+import { createWriteStream, promises as fs } from 'fs';
+import { parse } from 'path';
+import { pipeline } from 'stream/promises';
 
-function ensureImageData(imageData: ImageData): ImageData {
+const GRAY_CHANNEL_COUNT = 1;
+const RGB_CHANNEL_COUNT = 3;
+const RGBA_CHANNEL_COUNT = 4;
+
+/**
+ * Ensures that `imageData` is acceptable to `canvas`.
+ */
+export function ensureImageData(imageData: globalThis.ImageData): ImageData {
+  if (imageData instanceof ImageData) {
+    return imageData;
+  }
   return createImageData(imageData.data, imageData.width, imageData.height);
 }
 
+/**
+ * Loads a RAW image from a file.
+ */
+async function loadRawImage(
+  path: string,
+  width: number,
+  height: number,
+  channelCount: number
+): Promise<ImageData> {
+  assert(
+    channelCount === GRAY_CHANNEL_COUNT ||
+      channelCount === RGB_CHANNEL_COUNT ||
+      channelCount === RGBA_CHANNEL_COUNT
+  );
+
+  const imageData = createImageData(width, height);
+  const buffer = await fs.readFile(path);
+
+  for (
+    let srcOffset = 0, dstOffset = 0;
+    srcOffset < buffer.length;
+    srcOffset += channelCount, dstOffset += RGBA_CHANNEL_COUNT
+  ) {
+    imageData.data[dstOffset] = buffer[srcOffset];
+    imageData.data[dstOffset + 1] =
+      buffer[channelCount > GRAY_CHANNEL_COUNT ? srcOffset + 1 : srcOffset];
+    imageData.data[dstOffset + 2] =
+      buffer[channelCount > GRAY_CHANNEL_COUNT ? srcOffset + 2 : srcOffset];
+    imageData.data[dstOffset + 3] =
+      buffer[channelCount > RGB_CHANNEL_COUNT ? srcOffset + 3 : srcOffset];
+  }
+
+  return imageData;
+}
+
+/**
+ * Loads a RAW image from a file. The file name must be in the format
+ * `{label}-{width}x{height}-{bitsPerPixel}bpp.raw`. For example:
+ * `ballot-1700x2200-24bpp.raw` or `scan-1700x2200-8bpp.raw`.
+ */
+async function loadRawImageWithMetadataInFileName(
```

Comment:
FYI, this really belongs in `image-utils`. I moved it there in https://github.com/votingworks/vxsuite/pull/2828, but that hasn't been merged yet.

Reply:
Ah great, I was gonna ask you about that. Thanks for moving it.

Reply:
Ok updated here: [c9ce854](https://github.com/votingworks/vxsuite/pull/2826/commits/c9ce85424bf33f5d79bc792adecfdfe11ea71bcd)